pkg/router/template/router.go

```diff
@@ -450,11 +458,38 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 		}
 	}
 
+	var err error
```

Comment:
Why not implement this whole chunk as an admission layer?  Extended validations, performs checks, keeps a map of checked routes (resourceVersion+uid)